transform/app/api/routes/transform/batch.py

```diff
@@ -33,9 +34,20 @@ async def batch_update(
     data: dict | list[dict],
 ):
     """Transform a batch of a single type of data. Used for live update"""
+    task_ids = []
     if action == "delete":
         task = delete_data_by_id.delay(data_type, data, delete=True)
-        return {"task_id": task.id}
+        task_ids.append(task.id)
     else:
-        task = transform_batch.delay(data_type, data, full_update=False)
-        return {"task_id": task.id}
+        if data_type == "service":
+            task_ids.extend(update_service(data))
+        elif data_type == "data source":
+            task_ids.extend(update_data_source(data))
+        else:
+            task = transform_batch.delay(data_type, data, full_update=False)
+            task_ids.append(task.id)
+
+    if task_ids:
```

Comment:
just `return {"task_ids": task_ids}` is fine.
data_type & action are Literals so validation is already there

Reply:
Yes, you are right